sympy/sets/sets.py

```diff
@@ -917,7 +917,8 @@ def __new__(cls, start, end, left_open=False, right_open=False):
 
         inftys = [S.Infinity, S.NegativeInfinity]
         # Only allow real intervals (use symbols with 'is_extended_real=True').
-        if not all(i.is_extended_real is not False or i in inftys for i in (start, end)):
+        if not all(i.is_extended_real is not False or i in inftys
+            for i in (start, end, end - start)):
             raise ValueError("Non-real intervals are not supported")
```

Comment:
Which cases benefit from checking `end - start`?

Could use `fuzzy_and/or/not` here (and in other places).
```
if fuzzy_not(fuzzy_and(i.is_extended_real for s in (start, end, end-start))):
```
Tri-valued logic is confusing so I like to keep it as close as possible to normal logic:
```
if not all(is_real(i) for i in (start, end, end-start)):
```

Reply:
ok, changed.